### DISCUSSION TITLE
Optuna getting stuck

### DISCUSSION THREAD

**Discussion by kienerj** [optuna/optuna#5718] [Q&A]
As the title says. I create a study, start to run it and then it does a couple of trials, I haven't seen more than about 10 but usually it's less and then it gets stuck, just nothing happens anymore. But there is no error at all, just forever stuck. I then have to cancel and contiune as I'm saving the results to a DB.

Study is using in essence all default values:

    study = optuna.create_study(study_name=study_name, storage=storage_name, direction="maximize", load_if_exists=True)
    study.optimize(objective, n_trials=300)

This is running in jupyter lab but it shouldn't really matter right?

**Comment by nzw0301** [comment]
Hi, could you share minimal reproducible codes with us?

**Comment by kienerj** [comment]
The issue is I can't share the data I use plus I use optuna to optimize a deepchem model meaning you need very specific data to try and reproduce it (can't just generate random numbers).

I added the code in the attached file but as said not sure how useful it is without any data. 

[optuna_deepchem.txt](https://github.com/user-attachments/files/17507193/optuna_deepchem.txt)

**Reply by nzw0301** [reply]
Then could you generate some dummy data instead of your actual data in your code?

**Reply by kienerj** [reply]
I tried to repeat with a public data set but the issues does not happen there. So it does seem to be related to the dataset.

**Reply by nzw0301** [reply]
Thanks. Then I suspect that training given hyper-parameters takes too much time to be performed because the dataset and model are large.

**Reply by kienerj** [reply]
the public dataset is actually very large so I just took the first 2000 rows so I can actually test it in a reasonable time. I now increased it to 10k rows and it seems I'm no seeing the same behavior of the optimization just stopping. So it indeed may be related to model size but then 10k rows isn't really large at all?

**Reply by nzw0301** [reply]
Thanks. TBH, I don't know much about GCN in chem domain, but in vision or NLP, 10K samples sounds small. 

To investigate the issue more on your data instead of public dataset, how about interrupting your optimisation by executing `Ctrl-C` or something similar and looking at where your script is getting stuck?

**Comment by kienerj** [comment]
I made an example using a public dataset. Notebook attached.

The dataset can be found [here](https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/HIV.csv). This also gets stuck.

[Optuna Issue example.ipynb.txt](https://github.com/user-attachments/files/17571186/Optuna.Issue.example.ipynb.txt)
